File: java/move_zeroes.java
public class Solution {
    void swap(int[] nums, int i, int j) {
        int tmp=nums[i];
        nums[i]=nums[j];
        nums[j]=tmp;
    }
    
    public void moveZeroes(int[] nums) {
        for (int nonzero=0, i=0; i<nums.length; i++) {
            if (nums[i]!=0) {
                if (i!=nonzero) {
                    swap(nums, nonzero, i);
                }
                nonzero++;
            }
        }
    }
}

File: python/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n, end=len(nums), 0
        for i in range(n):
            if nums[i]!=0:
                if end!=i:
                    nums[end], nums[i]=nums[i], nums[end]
                end+=1

